gravitee-apim-console-webui/src/management/api/plans/list/api-plan-list.component.ts

```diff
@@ -246,6 +238,65 @@ export class ApiPlanListComponent implements OnInit, OnDestroy {
       .subscribe();
   }
 
+  private httpPlanDialog$(plan: Plan): Observable<Plan> {
+    return this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData>(GioConfirmDialogComponent, {
+        width: '500px',
+        data: {
+          title: `Publish plan`,
+          content: `Are you sure you want to publish the plan ${plan.name}?`,
+          confirmButton: `Publish`,
+        },
+        role: 'alertdialog',
+        id: 'publishPlanDialog',
+      })
+      .afterClosed()
+      .pipe(
+        filter((confirm) => confirm === true),
+        switchMap(() => this.plansService.publish(this.api.id, plan.id)),
+      );
+  }
+
+  private publishNativeKafkaPlan$(plan: Plan): Observable<Plan> {
+    return this.plansService.list(this.api.id, undefined, ['PUBLISHED'], undefined, 1, 9999).pipe(
+      switchMap((plansResponse) => {
+        const publishedKeylessPlan = plansResponse.data.filter((plan) => plan.security.type === 'KEY_LESS');
+        const publishedAuthPlans = plansResponse.data.filter((plan) => plan.security.type !== 'KEY_LESS');
+
+        if (plan.security?.type === 'KEY_LESS' && publishedAuthPlans.length) {
+          return this.nativeKafkaDialog$(plan, publishedAuthPlans);
+        } else if (plan.security?.type !== 'KEY_LESS' && publishedKeylessPlan.length) {
+          return this.nativeKafkaDialog$(plan, publishedKeylessPlan);
+        } else {
+          return this.httpPlanDialog$(plan);
+        }
+      }),
+      takeUntil(this.unsubscribe$),
+    );
+  }
+
+  private nativeKafkaDialog$(plan: Plan, plansToClose: Plan[]): Observable<Plan> {
+    const plansWithAuthentication = `plan${plansToClose.length > 1 ? 's' : ''} with authentication`;
+    const content = `Kafka APIs cannot have both plans with and without authentication published. Are you sure you want to publish the plan ${plan.name}? <br /><br />Your published ${plan.security.type === 'KEY_LESS' ? plansWithAuthentication : 'Keyless plan'} will be closed automatically.`;
```

Comment:
I think we need to check the behaviour with subscription. And indicate here that closing a plan with security will close the associated subscriptions. Like the message that closes a plan with security